src/measure_theory/simple_func_dense.lean

```diff
@@ -1,27 +1,39 @@
 /-
 Copyright (c) 2019 Zhouhang Zhou. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Zhouhang Zhou, Yury Kudryashov
+Authors: Zhouhang Zhou, Yury Kudryashov, Heather Macbeth
 -/
 import measure_theory.l1_space
 
 /-!
 # Density of simple functions
 
 Show that each Borel measurable function can be approximated,
-both pointwise and in `L¹` norm, by a sequence of simple functions.
+both pointwise and in `Lᵖ` norm, by a sequence of simple functions.
 
 ## Main definitions
 
 * `measure_theory.simple_func.nearest_pt (e : ℕ → α) (N : ℕ) : α →ₛ ℕ`: the `simple_func` sending
   each `x : α` to the point `e k` which is the nearest to `x` among `e 0`, ..., `e N`.
 * `measure_theory.simple_func.approx_on (f : β → α) (hf : measurable f) (s : set α) (y₀ : α)
   (h₀ : y₀ ∈ s) [separable_space s] (n : ℕ) : β →ₛ α` : a simple function that takes values in `s`
-  and approximates `f`. If `f x ∈ s`, then `measure_theory.simple_func.approx_on f hf s y₀ h₀ n x`
-  tends to `f x` as `n` tends to `∞`. If `α` is a `normed_group`, `f x - y₀`
-  is `measure_theory.integrable`, and `f x ∈ s` for a.e. `x`, then
-  `simple_func.approx_on f hf s y₀ h₀ n` tends to `f` in `L₁`. The main use case is `s = univ`,
-  `y₀ = 0`.
+  and approximates `f`.
+
+## Main results
+
+* `tendsto_approx_on` (pointwise convergence): If `f x ∈ s`, then the sequence of simple
+  approximations `measure_theory.simple_func.approx_on f hf s y₀ h₀ n`, evaluated at `x`,
+  tends to `f x` as `n` tends to `∞`.
+* `tendsto_approx_on_univ_Lp` (Lᵖ convergence): If `E` is a `normed_group` and `f` is measurable
+  and `mem_ℒp` (for `p < ∞`), then the simple functions `simple_func.approx_on f hf s 0 h₀ n` may
+  be considered as elements of `Lp E p μ`, and they tend in Lᵖ to `f`.
+* `tendsto_approx_on_univ_L1` (L¹ convergence): If `E` is a `normed_group` and `f` is measurable
+  and integrable, then the simple functions `simple_func.approx_on f hf s 0 h₀ n` may be considered
+  as elements of `Lp E 1 μ`, and they tend in L¹ to `f`.
+
+## TODO
+
+Simple functions are also dense in L^∞ -- prove this.
```

Comment:
This is not always true, even when the target space is second-countable -- for instance, the function from ℕ to a Hilbert space H mapping n to e_n (the n-th element in a Hilbert basis) is not approximated in L^∞ by simple functions. It's true when the target space is finite-dimensional, though.

Reply:
Indeed, I was starting to have my doubts as I thought about proving it :). I'll fix up this comment.